source/postprocess/visualization/devstress.cc

```diff
@@ -0,0 +1,146 @@
+
+        MaterialModel::MaterialModelInputs<dim> in(input_data,
+                                                   this->introspection());
+        MaterialModel::MaterialModelOutputs<dim> out(n_quadrature_points,
+                                                     this->n_compositional_fields());
+
+        // Compute the viscosity...
+        this->get_material_model().evaluate(in, out);
+
+        // ...and use it to compute the stresses
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          {
+            const SymmetricTensor<2,dim> strain_rate = in.strain_rate[q];
+            const SymmetricTensor<2,dim> compressible_strain_rate
+              = (this->get_material_model().is_compressible()
+                 ?
+                 strain_rate - 1./3 * trace(strain_rate) * unit_symmetric_tensor<dim>()
+                 :
+                 strain_rate);
+
+            const double eta = out.viscosities[q];
+
+            const SymmetricTensor<2,dim> devstress = 2*eta*compressible_strain_rate;
+            for (unsigned int i=0; i<SymmetricTensor<2,dim>::n_independent_components; ++i)
+              computed_quantities[q](i) = devstress[devstress.unrolled_to_component_indices(i)];
+          }
+      }
+
+
+      template <int dim>
+      std::vector<std::string>
+      DevStress<dim>::get_names () const
+      {
+        std::vector<std::string> names;
+        switch (dim)
+          {
+            case 2:
+              names.emplace_back("devstress_xx");
+              names.emplace_back("devstress_yy");
+              names.emplace_back("devstress_xy");
+              break;
+
+            case 3:
+              names.emplace_back("devstress_xx");
+              names.emplace_back("devstress_yy");
+              names.emplace_back("devstress_zz");
+              names.emplace_back("devstress_xy");
+              names.emplace_back("devstress_xz");
+              names.emplace_back("devstress_yz");
+              break;
+
+            default:
+              Assert (false, ExcNotImplemented());
+          }
+
+        return names;
+      }
+
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      DevStress<dim>::get_data_component_interpretation () const
+      {
+        return
+          std::vector<DataComponentInterpretation::DataComponentInterpretation>
+          (SymmetricTensor<2,dim>::n_independent_components,
+           DataComponentInterpretation::component_is_scalar);
+      }
+
+
+
+      template <int dim>
+      UpdateFlags
+      DevStress<dim>::get_needed_update_flags () const
+      {
+        return update_gradients | update_values | update_quadrature_points;
+      }
+
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(DevStress,
+                                                  "devstress",
```

Comment:
we typically spell out names, so please use "deviatoric stress"

Reply:
I changed the following files names: 
source/postprocess/visualization/devstress.cc --> source/postprocess/visualization/deviatoric_stress.cc and 
include/aspect/postprocess/visualization/devstress.h --> include/aspect/postprocess/visualization/deviatoric_stress.h

Furthermore: 
- Class name: DevStress to DeviatoricStress
- Postprocessing name in parameter file is now "deviatoric_stress"
- visualisation option names changed from "devstress_xx" to deviatoric_stress_xx", and similar for all other components of stress tensor.